### PR TITLE
Feature - Login view controller (UI).

### PR DESCRIPTION
**Pull request by josipas**
https://user-images.githubusercontent.com/52414221/177946316-d410f32a-ff2b-448a-ad01-35735b98edb6.mp4

### CODE REVIEW

File: QuizApp/QuizApp/LoginViewController.swift [hunk 1]
```diff
@@ -2,43 +2,146 @@ import UIKit
 
 class LoginViewController: UIViewController {
 
+    private let gradient = CAGradientLayer()
```
Comment (istaudacher) [inline QuizApp/QuizApp/LoginViewController.swift:5]: newline below

File: QuizApp/QuizApp/LoginViewController.swift [hunk 2]
```diff
@@ -2,43 +2,146 @@ import UIKit
...
         createViews()
         styleViews()
         defineLayoutForViews()
+        addActions()
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+
+        let startColor = UIColor(red: 0.453, green: 0.308, blue: 0.637, alpha: 1).cgColor
```
Comment (istaudacher) [inline QuizApp/QuizApp/LoginViewController.swift:34]: Please extract creating gradient in separate private function.

File: QuizApp/QuizApp/LoginViewController.swift [hunk 3]
```diff
@@ -2,43 +2,146 @@ import UIKit
...
+        case .password:
+            password = text
+        }
+
+        if
+            let email = email,
+            !email.isEmpty,
+            let password = password,
+            !password.isEmpty {
+                loginButton.isEnabled = true
```
Comment (istaudacher) [inline QuizApp/QuizApp/LoginViewController.swift]: Please extract styling loginButton in separate functions:
* enableLoginButton
* disableLoginButton

**Review by istaudacher — CHANGES_REQUESTED** [review]

**Review by istaudacher — APPROVED** [review]

File: QuizApp/QuizApp/LoginViewController.swift [hunk 4]
```diff
@@ -2,43 +2,159 @@ import UIKit
...
+
+        if
+            let email = email,
+            !email.isEmpty,
+            let password = password,
+            !password.isEmpty {
+                enableLoginButton()
+        } else {
+            disableLoginButton()
+        }
```
Comment (marinzg) [inline QuizApp/QuizApp/LoginViewController.swift]: `{` goes into new line - [style guide](https://github.com/fiveagency/ios-swift-guide/tree/master/Code%20Style#if-else-statement)

**Review by marinzg — CHANGES_REQUESTED** [review]

**Review by istaudacher — APPROVED** [review]

**Review by marinzg — APPROVED** [review]

**Pull request merged**